fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/leaderboard/TimeWindowLeaderboardIndexMaintainer.java

```diff
@@ -589,10 +589,8 @@ public CompletableFuture<Void> deleteWhere(Transaction tr, @Nonnull Tuple prefix
                         final Tuple leaderboardGroupKey = leaderboard.getSubspaceKey().addAll(prefix);
                         // NOTE: Range.startsWith(), Subspace.range() and so on cover keys *strictly* within the range, but we
                         // may store something at the group root as well.
-                        final byte[] indexKey = indexSubspace.pack(leaderboardGroupKey);
-                        tr.clear(indexKey, ByteArrayUtil.strinc(indexKey));
```

Comment:
Note that this is `ByteArrayUtil.strinc`, not `ByteArrayUtil.keyAfter`. These really are non-trivial ranges, as explained by the comment.

Reply:
Let me take a closer look.